### PR TITLE
gh-106368: Clean up Argument Clinic tests

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-106368

### CODE REVIEW

**Comment by erlend-aasland** [comment]
If we're going to increase coverage for Argument Clinic, we need to clean up the test code a little bit; we need readable tests, and we need to be able to debug existing tests without pulling our hair.

File: Lib/test/test_clinic.py [hunk 1]
```diff
@@ -772,46 +857,39 @@ def test_directive(self):
...
+        expected_failure_message = (
+            "Error on line 0:\n"
+            "Annotations must be either a name, a function call, or a string.\n"
         )
-        self.assertEqual(s, expected_failure_message)
-
-        s = self.parse_function_should_fail(
-            'module os\nos.access\n   path: ["list", "of", "strings"]'
+        dataset = (
+            'module os\nos.access\n   path: 42',
```
Comment (sobolevn) [inline Lib/test/test_clinic.py:869]: I would prefer to have these as multi-line strings.

File: Lib/test/test_clinic.py [hunk 2]
```diff
@@ -288,257 +289,333 @@ def test_clinic_1(self):
...
+              y
+                Y-coordinate.
+              x
+                X-coordinate.
+              ch
+                Character to add.
+              attr
+                Attributes for the character.
+        """).strip()
+        self.assertEqual(function.docstring.strip(), expected)
```
Comment (sobolevn) [inline Lib/test/test_clinic.py]: We can make a custom `assert` method that does all the `dedent` + `strip` stuff, it is used quite a lot in this module.

**Review by sobolevn — COMMENTED** [review]
Thanks for doing this, in my opinion, it is much more readable now.

File: Lib/test/test_clinic.py (hunk 1 above)
Comment (erlend-aasland) [reply Lib/test/test_clinic.py:869]: Ideally, yes, but for parametrised tests like this, it is ok to have the input lines as a single strings, especially given they are so short.

File: Lib/test/test_clinic.py (hunk 2 above)
Comment (erlend-aasland) [reply Lib/test/test_clinic.py]: Good idea. I added a checkDocstring helper.

**Review by sobolevn — APPROVED** [review]
👍

**Comment by erlend-aasland** [comment]
After gh-106373, we should make the `test_disallowed_*` tests more accurate. Most of them simply run `self.parse_function_should_fail` without checking the error message. This makes it hard to verify that the test is checking the right path.

**Pull request merged**